internal/providers/pluginfw/resources/sharing/resource_share.go

```diff
@@ -0,0 +1,395 @@
+			Action:     sharing.SharedDataObjectUpdateAction(action),
+			DataObject: &obj,
+		},
+		)
+	}
+	return sharing.UpdateShare{
+		Name:    si.Name,
+		Owner:   si.Owner,
+		Updates: changes,
+	}
+}
+
+type ShareResource struct {
+	Client *common.DatabricksClient
+}
+
+func (r *ShareResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = pluginfwcommon.GetDatabricksStagingName(resourceName)
+}
+
+func (r *ShareResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	attrs, blocks := tfschema.ResourceStructToSchemaMap(ShareInfoExtended{}, func(c tfschema.CustomizableSchema) tfschema.CustomizableSchema {
+		c.SetRequired("name")
+
+		c.AddPlanModifier(stringplanmodifier.RequiresReplace(), "name") // ForceNew
+		c.AddPlanModifier(int64planmodifier.UseStateForUnknown(), "created_at")
+		c.AddPlanModifier(stringplanmodifier.UseStateForUnknown(), "created_by")
+		c.AddPlanModifier(int64planmodifier.UseStateForUnknown(), "object", "added_at")
+		c.AddPlanModifier(stringplanmodifier.UseStateForUnknown(), "object", "added_by")
+
+		c.SetRequired("object", "data_object_type")
+		c.SetRequired("object", "partitions", "values", "op")
+		c.SetRequired("object", "partitions", "values", "name")
+		return c
+	})
+	resp.Schema = schema.Schema{
+		Description: "Terraform schema for Databricks Share",
+		Attributes:  attrs,
+		Blocks:      blocks,
+	}
+}
+
+func (d *ShareResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if d.Client == nil && req.ProviderData != nil {
+		d.Client = pluginfwcommon.ConfigureResource(req, resp)
+	}
+}
+
+func (r *ShareResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	w, diags := r.Client.GetWorkspaceClient()
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	var plan ShareInfoExtended
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	var planGoSDK sharing.ShareInfo
+	resp.Diagnostics.Append(converters.TfSdkToGoSdkStruct(ctx, plan, &planGoSDK)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	var createShare sharing.CreateShare
+	resp.Diagnostics.Append(converters.TfSdkToGoSdkStruct(ctx, plan, &createShare)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	shareInfo, err := w.Shares.Create(ctx, createShare)
+	if err != nil {
+		resp.Diagnostics.AddError("failed to create share", err.Error())
+		return
+	}
+
+	shareChanges := shareChanges(planGoSDK, string(sharing.SharedDataObjectUpdateActionAdd))
+
+	updatedShareInfo, err := w.Shares.Update(ctx, shareChanges)
+	if err != nil {
+		// delete orphaned share if update fails
+		if d_err := w.Shares.DeleteByName(ctx, shareInfo.Name); d_err != nil {
+			resp.Diagnostics.AddError("failed to delete orphaned share", d_err.Error())
```

Comment:
We should also add err along with d_err in the diagnostics here as we return so line 205 won't execute.